### PR TITLE
Add QubitPermutationGate and tuple input support to ClassicalStateSimulator

### PR DESCRIPTION
**Pull request by Aqil-Ahmad**
## Summary
This PR enhances the `ClassicalStateSimulator` by adding support for `QubitPermutationGate` operations and enabling `tuple[int]` input for the initial state parameter.

Fixes : #7566 

## Changes Made
- **QubitPermutationGate Support**: Implemented handling of qubit permutation operations in `_act_on_fallback_` method
- **Tuple Input Support**: Added support for `tuple[int, ...]` as initial state input alongside existing `int`, `list[int]`, and `np.ndarray` options
- **Documentation Updates**: Updated docstrings and error messages
- **Testing**: Added test coverage for:
  - QubitPermutationGate operations with various permutations
  - Tuple input validation and functionality
  - Edge cases and error conditions

## Testing

- [x]  All existing tests pass
- [x]  New tests for QubitPermutationGate functionality and tuple input handling

### CODE REVIEW

File: cirq-core/cirq/sim/classical_simulator.py [hunk 1]
```diff
@@ -77,20 +77,24 @@ class ClassicalBasisSimState(SimulationState[ClassicalBasisState]):
 
     def __init__(
         self,
-        initial_state: int | list[int] = 0,
+        initial_state: int | list[int] | tuple[int, ...] = 0,
```
Comment (NoureldinYosri) [inline cirq-core/cirq/sim/classical_simulator.py]:
```suggestion
        initial_state: int | Sequence[int] = 0,
```

File: cirq-core/cirq/sim/classical_simulator.py [hunk 2]
```diff
@@ -77,20 +77,24 @@ class ClassicalBasisSimState(SimulationState[ClassicalBasisState]):
...
         qubits: Sequence[cirq.Qid] | None = None,
         classical_data: cirq.ClassicalDataStore | None = None,
     ):
         """Initializes the ClassicalBasisSimState object.
 
         Args:
             qubits: The qubits to simulate.
-            initial_state: The initial state for the simulation.
+            initial_state: The initial state for the simulation. Accepts int, list[int],
```
Comment (NoureldinYosri) [inline cirq-core/cirq/sim/classical_simulator.py]:
```suggestion
            initial_state: The initial state for the simulation. Accepts int or a sequence of int.
```

File: cirq-core/cirq/sim/classical_simulator.py [hunk 3]
```diff
@@ -100,10 +103,16 @@ def __init__(
...
+            if initial_state.ndim != 1:
+                raise ValueError(
+                    f'initial_state must be 1-dimensional, got shape {initial_state.shape}'
+                )
+            state = ClassicalBasisState(list(initial_state))
+        elif isinstance(initial_state, (list, tuple)):
+            state = ClassicalBasisState(list(initial_state))
         else:
-            raise ValueError('initial_state must be an int or list[int] or np.ndarray')
+            raise ValueError('initial_state must be an int, list[int], tuple[int], or np.ndarray')
```
Comment (NoureldinYosri) [inline cirq-core/cirq/sim/classical_simulator.py]: Sequence[
```suggestion
            raise ValueError('initial_state must be an int or Sequence[int]')
```

File: cirq-core/cirq/sim/classical_simulator_test.py [hunk 4]
```diff
@@ -60,6 +60,15 @@ def test_Swap():
     np.testing.assert_equal(results, expected_results)
 
 
+def test_qubit_permutation_gate():
```
Comment (NoureldinYosri) [inline cirq-core/cirq/sim/classical_simulator_test.py]:
```suggestion
@pytest.mark.parameterize(['n', 'perm', 'state'], [
    (n, np.random.permutation(n).tolist(), np.random.choice(2, size=n))               for n in np.random.randint(3, 8, size=10)
])
def test_qubit_permutation_gate(n, perm, state):
     inv_perm = [perm.index(i) for i in range(n)]
     target = [state[p] for p in inv_perm]
     ...
     np.testing.assert_equal(result.measurements['key'], target)
```

**Review by NoureldinYosri — CHANGES_REQUESTED** [review]
please check the definition of the permutation gate https://quantumai.google/reference/python/cirq/QubitPermutationGate


and make sure the ClassicalStateSimulator is consistent with the cirq.Simulator

File: cirq-core/cirq/sim/classical_simulator_test.py [hunk 5]
```diff
@@ -60,6 +60,23 @@ def test_Swap():
...
+        for n in np.random.randint(3, 8, size=10)
+    ],
+)
+def test_qubit_permutation_gate(n, perm, state):
+    qubits = cirq.LineQubit.range(n)
+    perm_gate = cirq.QubitPermutationGate(perm)
+    circuit = cirq.Circuit(perm_gate(*qubits), cirq.measure(*qubits, key='key'))
+    sim = cirq.ClassicalStateSimulator()
+    result = sim.simulate(circuit, initial_state=state)
+    expected = [state[perm[i]] for i in range(n)]
```
Comment (NoureldinYosri) [inline cirq-core/cirq/sim/classical_simulator_test.py]: this doesn't match the action of the state vector simulator (cirq.Simulator). if you believe the state vector simulator is the wrong one please file a bug, otherwise please make them match

**Review by NoureldinYosri — APPROVED** [review]
thanks
